src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -332,43 +336,39 @@ public override void VisitParameterDeclarationSyntax(ParameterDeclarationSyntax
                 }
 
                 return assignedType;
+            });
 
-                /*
-                 * Calling FirstOrDefault here because when there are duplicate decorators,
-                 * the top most one takes precedence.
-                 */
-                DecoratorSyntax? GetDecorator(string decoratorName) => syntax.Decorators.FirstOrDefault(decoratorSyntax =>
+        public override void VisitImportDeclarationSyntax(ImportDeclarationSyntax syntax)
+            => AssignTypeWithDiagnostics(syntax, diagnostics =>
+            {
+                if (binder.GetSymbolInfo(syntax) is not ImportedNamespaceSymbol namespaceSymbol)
+                {
+                    return ErrorType.Empty();
```

Comment:
>Empty [](http://example.com/codeflow?start=37&length=5)

This path would be executed when we have parse errors, right? If yes, can you add a comment?

Reply:
Yup, done.